backend/infrahub/graphql/mutations/menu.py

```diff
@@ -0,0 +1,103 @@
+from typing import TYPE_CHECKING, Any, Optional
+
+from graphene import InputObjectType, Mutation
+from graphql import GraphQLResolveInfo
+from typing_extensions import Self
+
+from infrahub.core.branch import Branch
+from infrahub.core.constants import RESTRICTED_NAMESPACES
+from infrahub.core.manager import NodeManager
+from infrahub.core.node import Node
+from infrahub.core.protocols import CoreMenuItem
+from infrahub.core.schema import NodeSchema
+from infrahub.database import InfrahubDatabase
+from infrahub.exceptions import ValidationError
+from infrahub.graphql.mutations.main import InfrahubMutationMixin
+
+from .main import InfrahubMutationOptions
+
+if TYPE_CHECKING:
+    from infrahub.graphql.initialization import GraphqlContext
+
+EXTENDED_RESTRICTED_NAMESPACES = RESTRICTED_NAMESPACES + ["Builtin"]
+
+
+def validate_namespace(data: InputObjectType) -> None:
+    namespace = data.get("namespace")
+    if isinstance(namespace, dict) and "value" in namespace:
+        namespace_value = str(namespace.get("value"))
+        if namespace_value.title() in EXTENDED_RESTRICTED_NAMESPACES:
```

Comment:
Just a quick comment here. I think we should remove this `.title()` call and instead add a regex to the module that ensures that the `name` and `namespace` field follows the correct convention, we could use `NAMESPACE_REGEX` and `NODE_KIND_REGEX`.